app/activeproject.cpp

```diff
@@ -465,31 +450,28 @@ void ActiveProject::setMapTheme( const QString &themeName )
 
 void ActiveProject::updateActiveLayer()
 {
+  QList< QgsMapLayer * > visibleLayers = getVisibleLayers();
+
+  if ( !mQgsProject || visibleLayers.isEmpty() )
+    return;
+
   if ( !InputUtils::layerVisible( mActiveLayer.layer(), mQgsProject ) )
```

Comment:
We already have `visibleLayers` here, so we could directly check if it contains `mActiveLayer.layer()`